sys-devel/llvm-common/llvm-common-14.0.6.ebuild

```diff
@@ -1,28 +1,70 @@
-# Copyright 1999-2022 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-inherit llvm.org
+inherit elisp-common llvm.org
 
 DESCRIPTION="Common files shared between multiple slots of LLVM"
 HOMEPAGE="https://llvm.org/"
 
 LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA"
 SLOT="0"
 KEYWORDS="amd64 arm arm64 ~ppc ppc64 ~riscv sparc x86 ~amd64-linux ~ppc-macos ~x64-macos"
+IUSE="emacs"
 
 RDEPEND="
 	!sys-devel/llvm:0
 "
+BDEPEND="
+	emacs? ( app-editors/emacs )
+"
 
-LLVM_COMPONENTS=( llvm/utils/vim )
+LLVM_COMPONENTS=( llvm/utils )
 llvm.org_set_globals
 
+src_compile() {
+	default
+
+	if use emacs ; then
+		elisp-compile "${S}" emacs/llvm-mode.el
+		elisp-compile "${S}" emacs/tablegen-mode.el
+	fi
+}
+
 src_install() {
 	insinto /usr/share/vim/vimfiles
-	doins -r */
+	doins -r vim/*/
 	# some users may find it useful
-	newdoc README README.vim
-	dodoc vimrc
+	newdoc vim/README README.vim
+	dodoc vim/vimrc
+
+	if use emacs ; then
+		elisp-install llvm-mode emacs/llvm-mode.el
+		elisp-install tablegen-mode emacs/tablegen-mode.el
+
+		cat <<EOF >> 50llvm-mode-gentoo.el
+;;; llvm-mode site-lisp configuration
+
```

Comment:
The comment isn't needed, `elisp-site-file-install` will add it automatically.

Reply:
Will fix these things too once your elisp-make-site-file PR gets merged

Reply:
nvm, it was already merged, fixing soon